### PR TITLE
Update Ireland holidays: St. Bridget's Day

### PR DESCRIPTION
**Pull request by PPsyrius**
Due to the code changes for `beta` introduced in #841, #920 's patch no longer works.

This PR is aimed to keep #917's logic as stated below:
>The current code (ireland.py) marks 1st February as a "holiday", and 6th February as the Observed Holiday (2023).
>
> 1st February is the saints day but is not the Holiday. The 6th February is the Holiday (1st Monday in February), unless the 1st falls on a Friday.

### CODE REVIEW

File: holidays/countries/ireland.py [hunk 1]
```diff
@@ -41,9 +41,7 @@ def _populate(self, year):
         if year >= 2023:
             name = "St. Brigid's Day"
             dt = date(year, FEB, 1)
-            self[dt] = name
-            if self.observed and not self._is_friday(dt):
-                self[dt + rd(weekday=MO)] = name + " (Observed)"
+            self[dt if self._is_friday(dt) else dt + rd(weekday=MO)] = name
```
Comment (KJhellico) [inline holidays/countries/ireland.py:44]:
```
        if year >= 2023:
            dt = date(year, FEB, 1)
            self[
                dt if self._is_friday(dt) else dt + rd(weekday=MO)
            ] = "St. Brigid's Day"

```

**Review by arkid15r — COMMENTED** [review]
LGTM

Any chance to have this reflected in the tests please? :)

**Comment by PPsyrius** [comment]
I've added test case for Ireland's St. Brigid's Day from 2023-2032, should be good to go IMO.

**Comment by arkid15r** [comment]
> I've added test case for Ireland's St. Brigid's Day from 2023-2032, should be good to go IMO.

I appreciate that a lot!

**Review by arkid15r — COMMENTED** [review]
LGTM

**Comment by arkid15r** [comment]
Thanks for wrapping this up @PPsyrius!

**Pull request merged**